### PR TITLE
Add Devise Authentication and Database Migrations

### PR DESCRIPTION
**Pull request by lerfast**
Hellow fellow coding partners, I did the following tasks:

## Description

This pull request adds Devise for user authentication in the Rails backend. It includes the necessary database migrations to support Devise's user model.

## Changes

- Installed Devise gem.
- Configured Devise in the User model.
- Added database migrations for Devise, including fields for encrypted_password, reset_password, and rememberable.

## Testing

- Ran migrations successfully.
- Tested user sign-up and login functionality through Rails console and Postman.

## Why

Implementing Devise provides a secure and scalable way to handle user authentication, which is essential for the user management feature of our application.

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  # ==> Navigation configuration
+  # Lists the formats that should be treated as navigational. Formats like
+  # :html should redirect to the sign in page when the user does not have
+  # access, but formats like :xml or :json, should return 401.
+  #
+  # If you have any extra navigational formats, like :iphone or :mobile, you
+  # should add them to the navigational formats lists.
+  #
+  # The "*/*" below is required to match Internet Explorer requests.
+  # config.navigational_formats = ['*/*', :html, :turbo_stream]
```
Comment (cancelei) [inline config/initializers/devise.rb:266]: In the link I sent you, the developer mentions that this line should the commented in as a blank array. We need to remember this, and come back to this line if we face any issues later on.

**Review by cancelei — APPROVED** [review]
Great Code! I saw the video you created with the 4 min demo of this feature!

File: config/initializers/devise.rb (hunk 1 above)
Comment (cancelei) [reply config/initializers/devise.rb:266]: https://medium.com/@alessandrahagarty/using-devise-for-authentication-in-a-react-rails-app-f6a0eb87fbd5

**Comment by alvp01** [comment]
I tested, and it is as you say in the video, we should make some validations. But the initial feature is working

**Review by alvp01 — APPROVED** [review]
Good for the first Iteration

**Review by Gardimy — APPROVED** [review]
Well done setup the devise...✨✨✔🎇👍

**Pull request merged**